libraries/Microsoft.Bot.Builder/Streaming/StreamingRequestHandler.cs

```diff
@@ -381,11 +383,24 @@ private async Task ReconnectAsync(IDictionary<string, string> requestHeaders = n
                 }
             }
 
+            // Set the authentication header if it wasn't passed in.
+            if (requestHeaders != null && !requestHeaders.ContainsKey("authentication") && !requestHeaders.ContainsKey("Authentication"))
+            {
+                try
+                {
+                    clientWebSocket.Options.SetRequestHeader("authentication", await _appCredentials.GetTokenAsync().ConfigureAwait(false));
+                }
+                catch (Exception e)
+                {
+                    _logger.LogError(e.Message);
```

Comment:
Is not throwing here the right behavior? That would still try to connect to the web socket on line 403, not sure if that sounds right

Reply:
Good point, in the reconnect scenario it's appropriate for the bot to decide to abort.